addons/google_calendar/utils/google_event.py

```diff
@@ -65,8 +66,10 @@ def ids(self):
 
     @property
     def rrule(self):
-        if self.recurrence and 'RRULE:' in self.recurrence[0]:  # LUL TODO what if there are something else in the list?
-            return self.recurrence[0][6:]  # skip "RRULE:" in the rrule string
+        if self.recurrence:
+            # Find the rrule in the list
+            rrule = [rr for rr in self.recurrence if 'RRULE:' in rr][0]
```

Comment:
Are we sure we always have a RRULE:blah in self.recurrence (aka, could[0] crash somehow) ? As I don't know the internals of google calendar, just asking.

Reply:
I think so https://developers.google.com/calendar/concepts/events-calendars#recurrence_rule 
I did not notice any recurrence without rrule in the entire data we fetch from our employee calendar. 

If it's not the case and we handle the case by returning False, None etc.. it crashes here https://github.com/odoo/odoo/blob/14.0/addons/calendar/models/calendar_recurrence.py#L461  I think it's more clear to crash in this part of the code. We could maybe raise a even better error

But I'm having hard time finding a recurrent event without rrule

Reply:
You could also use a next in order to keep your :monocle_face: .